src/main/java/cn/edu/tsinghua/tsfile/common/conf/TSFileDescriptor.java

```diff
@@ -1,88 +1,124 @@
 import java.util.Properties;
+import java.util.Set;
 
 /**
- * TSFileDescriptor is used to load TSFileConfig and provide configure information
+ * TSFileDescriptor is used to load TSFileConfig and provide configure
+ * information
  *
  * @author kangrong
  */
 public class TSFileDescriptor {
-    private static final Logger LOGGER = LoggerFactory.getLogger(TSFileDescriptor.class);
-    
-	private static class TsfileDescriptorHolder{
+	private static final Logger LOGGER = LoggerFactory.getLogger(TSFileDescriptor.class);
+
+	private static class TsfileDescriptorHolder {
 		private static final TSFileDescriptor INSTANCE = new TSFileDescriptor();
 	}
-    
-    private TSFileConfig conf = new TSFileConfig();
 
-    private TSFileDescriptor() {
-        loadProps();
-    }
+	private TSFileConfig conf = new TSFileConfig();
+
+	private TSFileDescriptor() {
+		loadProps();
+	}
+
+	public static final TSFileDescriptor getInstance() {
+		return TsfileDescriptorHolder.INSTANCE;
+	}
+
+	public TSFileConfig getConfig() {
+		return conf;
+	}
+
+	private void multiplicityWarning(String resource, ClassLoader classLoader) {
+		try {
+			Set<URL> urlSet = Loader.getResources(resource, classLoader);
+			if (urlSet != null && urlSet.size() > 1) {
+				LOGGER.warn("Resource [{}] occurs multiple times on the classpath", resource);
+				for (URL url : urlSet) {
+					LOGGER.warn("Resource [{}] occurs at [{}]", resource, url.toString());
+				}
+			}
+		} catch (IOException e) {
+		}
+	}
+	
+	private URL getResource(String filename, ClassLoader classLoader){
+		return Loader.getResource(filename, classLoader);
+	}
 
-    public static final TSFileDescriptor getInstance() {
-        return TsfileDescriptorHolder.INSTANCE;
-    }
+	/**
+	 * load an .properties file and set TSFileConfig variables
+	 */
+	private void loadProps() {
+		InputStream inputStream = null;
+		String url = System.getProperty(SystemConstant.TSFILE_CONF, null);
+		if (url == null) {
+			url = System.getProperty(SystemConstant.TSFILE_HOME, null);
+			if (url != null) {
+				url = url + File.separator + "conf" + File.separator + TSFileConfig.CONFIG_NAME;
+			} else {
+				ClassLoader  classLoader = Loader.getClassLoaderOfObject(this);
+				URL u = getResource(TSFileConfig.CONFIG_NAME, classLoader);
+				if(u == null){
+					LOGGER.warn("Failed to find config file {} at classpath, use default configuration", TSFileConfig.CONFIG_NAME);
+					return;
+				}
+				else{
+					multiplicityWarning(TSFileConfig.CONFIG_NAME, classLoader);
+					url = u.getFile();
+				}
+			}
+		}
+		try {
+			inputStream = new FileInputStream(new File(url));
+		} catch (FileNotFoundException e) {
+			LOGGER.warn("Fail to find config file {}", url);
+			return;
+		}
 
-    public TSFileConfig getConfig() {
-        return conf;
-    }
+		LOGGER.info("Start to read config file {}", url);
+		Properties properties = new Properties();
+		try {
+			properties.load(inputStream);
+			conf.groupSizeInByte = Integer

```

Comment:
Here we explicitly load some parameters like .etc, It's not scalable if new parameters appear. why we don't use the YAML importer or other public utils?